ci/images/plugin-compiler/Dockerfile

```diff
@@ -21,7 +21,7 @@ RUN  apt-get remove -y --allow-remove-essential --auto-remove mercurial \
 
 
 COPY . $TYK_GW_PATH
-RUN cd $TYK_GW_PATH && go mod vendor
+RUN cd $TYK_GW_PATH && go mod tidy && go mod vendor
```

Comment:
I cannot reproduce this locally with `GO111MODULE=on` which indicates that the workaround in the _Fix vendor_ step might be the problem. Local build log below:
```
$ docker run --rm --privileged -e GITHUB_TOKEN=xoxoxo -e GOPRIVATE=github.com/TykTechnologies -e PKG_SIGNING_KEY="sekret" -e NFPM_STD_PASSPHRASE=sekret-e NFPM_PAYG_PASSPHRASE=sekret -e GPG_FINGERPRINT=12B5D62C28F57592D1575BD51ED14C59E37DAC20 -e GPG_KEYID=A924CCFFB430E620F35D743D802F97F6391AC324 -e GOFLAGS=-buildvcs=false -v /var/run/docker.sock:/var/run/docker.sock -v ~/.docker/config.json:/root/.docker/config.json -v /home/alok/go/src/github.com/TykTechnologies/tyk:/go/tyk -w /go/tyk -e GO111MODULE=on -e CGO_ENABLED=1 -e GOLANG_CROSS=1.19-bullseye tykio/golang-cross:1.19-bullseye bash -c "ci/bin/unlock-agent.sh && goreleaser --clean --snapshot -f ci/goreleaser/goreleaser.yml"
Configuring gpg-agent to accept a passphrase
gpg-connect-agent: no running gpg-agent - starting '/usr/bin/gpg-agent'
gpg-connect-agent: waiting for the agent to come up ... (5s)
gpg-connect-agent: connection to agent established
OK
OK
gpg-connect-agent: closing connection to agent
Configuring gpg not to look for a tty
gpg: keybox '/root/.gnupg/pubring.kbx' created
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key 802F97F6391AC324: public key "Team Tyk (package signing) <team@tyk.io>" imported
gpg: key 802F97F6391AC324: secret key imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg:       secret keys read: 1
gpg:   secret keys imported: 1
  • starting release...
  • loading config file                              file=ci/goreleaser/goreleaser.yml
  • loading environment variables
    • using token from "$GITHUB_TOKEN"
  • getting and validating git state
    • accepting to run without a git repository because this is a snapshot
    • building...                                    commit=none latest tag=v0.0.0
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
      • DEPRECATED: `archives.rlcp` will be the default soon, check https://goreleaser.com/deprecations#archivesrlcp for more info
      • DEPRECATED: `nfpms.replacements` should not be used anymore, check https://goreleaser.com/deprecations#nfpmsreplacements for more info
      • DEPRECATED: `nfpms.replacements` should not be used anymore, check https://goreleaser.com/deprecations#nfpmsreplacements for more info
  • snapshotting
    • building snapshot...                           version=0.0.0-SNAPSHOT-none
  • checking distribution directory
    • cleaning dist
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/std-arm64_linux_arm64/tyk
    • building                                       binary=dist/std-linux_linux_amd64_v1/tyk
    • took: 3m40s
  • archives
    • creating                                       archive=dist/tyk_0.0.0-SNAPSHOT-none_linux_amd64.tar.gz
    • creating                                       archive=dist/tyk_0.0.0-SNAPSHOT-none_linux_arm64.tar.gz
    • took: 41s
  • linux packages
    • creating                                       arch=amd64v1 file=dist/tyk-gateway-0.0.0~SNAPSHOT~none-1.x86_64.rpm format=rpm package=tyk-gateway
    • creating                                       arch=arm64 file=dist/tyk-gateway_0.0.0~SNAPSHOT~none_arm64.deb format=deb package=tyk-gateway
    • creating                                       arch=amd64v1 file=dist/tyk-gateway_0.0.0~SNAPSHOT~none_amd64.deb format=deb package=tyk-gateway
    • creating                                       arch=arm64 file=dist/tyk-gateway-0.0.0~SNAPSHOT~none-1.aarch64.rpm format=rpm package=tyk-gateway
    • took: 9s
  • docker images
    • building docker image                          image=tykio/tyk-hybrid-docker:v0.0.0-arm64
    • building docker image                          image=tykio/tyk-hybrid-docker:v0.0.0-amd64
    • building docker image                          image=tykio/tyk-gateway:v0.0.0-arm64
    • building docker image                          image=tykio/tyk-plugin-compiler:v0.0.0
    • building docker image                          image=tykio/tyk-gateway:v0.0.0-amd64
    • took: 11m58s
  • storing release metadata
    • writing                                        file=dist/artifacts.json
    • writing                                        file=dist/metadata.json
  • you are using deprecated options, check the output above for details
  • release succeeded after 16m28s
```